### PR TITLE
Tests for the Sign Up use case. 100% coverage and all tests pass

### PR DESCRIPTION
**Pull request by stefanateljevic04**
Screen Shot 2023-11-19 at 11 43 25 AM

Attached screenshot of coverage. @merrickliu888 @jackkfan0305 @samaslanowicz @vrindasubhash please review and let me know if any changes are needed

### CODE REVIEW

File: test/use_case/signup/SignupInteractorTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                assertEquals("newUser", user.getUsername());
+                assertTrue(userRepository.userExists("newUser"));
+            }
+
+            @Override
+            public void prepareFailView(String error) {
+                fail("Use case failure is unexpected.");
+            }
+        };
+        failurePresenter = new SignupOutputBoundary() {
```
Comment (merrickliu888) [inline test/use_case/signup/SignupInteractorTest.java:38]: This part isn't necessary. You instantiate them later.

File: test/use_case/signup/SignupInteractorTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    private CommonUserFactory userFactory;
+    private SignupInteractor interactor;
+    private SignupOutputBoundary successPresenter;
+    private SignupOutputBoundary failurePresenter;
+
+    @Before
+    public void setUp() {
+        userRepository = new MemoryUserDataAccessObject();
+        userFactory = new CommonUserFactory();
+        successPresenter = new SignupOutputBoundary() {
```
Comment (merrickliu888) [inline test/use_case/signup/SignupInteractorTest.java:26]: Move this down into your success test.

File: test/use_case/signup/SignupInteractorTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            public void prepareSuccessView(SignupOutputData user) {
+                fail("Use case success is unexpected.");
+            }
+
+            @Override
+            public void prepareFailView(String error) {
+                // Error message will be set in individual tests
+            }
+        };
+        interactor = new SignupInteractor(userRepository, successPresenter, userFactory);
```
Comment (merrickliu888) [inline test/use_case/signup/SignupInteractorTest.java:49]: Move this down into success test.

File: test/use_case/signup/SignupInteractorTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    @After
+    public void tearDown() {
```
Comment (merrickliu888) [inline test/use_case/signup/SignupInteractorTest.java:53]: Not necessary.

**Review by merrickliu888 — CHANGES_REQUESTED** [review]
Looks good. Just some small refactoring.

**Comment by samaslanowicz** [comment]
Did you test the get methods in the input data and output data files too? Add those into this PR once done

**Comment by samaslanowicz** [comment]
nvm

**Review by jackkfan0305 — APPROVED** [review]
Everything looks good

**Review by merrickliu888 — APPROVED** [review]
Good fixes!

**Pull request merged**